nikola/plugins/basic_import.py

```diff
@@ -140,6 +140,24 @@ def write_content(cls, filename, content, rewrite_html=True):
         with open(filename, "wb+") as fd:
             fd.write(content)
 
+    @classmethod
```

Comment:
Why bother? A class method should only be used for alternate constructors
that return instances of the class (eg. datetime.datetime.fromtimestamp).
Other things should be regular methods. Static methods make sense
sometimes, but a regular method is enough.

-- 
Chris Warrick <https://chriswarrick.com/>
Sent from my Galaxy S3.
On 8 Mar 2016 23:36, "Felix Fontein" <notifications@github.com> wrote:

> In nikola/plugins/basic_import.py
> <https://github.com/getnikola/nikola/pull/2292#discussion_r55441024>:
>
> > @@ -140,6 +140,24 @@ def write_content(cls, filename, content, rewrite_html=True):
> >          with open(filename, "wb+") as fd:
> >              fd.write(content)
> >
> > +    @classmethod
>
> Out of curiosity: what's so bad about making something @classmethod in
> Python?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/getnikola/nikola/pull/2292/files#r55441024>.
>